server/ws/websockets.go

```diff
@@ -24,10 +24,15 @@ type Hub interface {
 	SetReceiveWSMessage(func(data []byte))
 }
 
+type Listener struct {
+	Lock *sync.RWMutex
+	Conn *websocket.Conn
```

Comment:
I was thinking about embedding it, but I guess there is not a problem about using in this way, and probably it is just simpler.

Reply:
👍